rust/Cargo.toml

```diff
@@ -127,10 +130,18 @@ prost-types = { version = "0.11", default-features = false }
 num-bigint = { version = "0.4.5", default-features = false }
 num-integer = { version = "0.1.46", default-features = false }
 num-traits = { version = "0.2.19", default-features = false }
+blake2b_simd = { version = "1.0.2", default-features = false }
+getrandom = "0.2"
 # third party dependencies end
 
-[patch.crates-io.bitcoin]
-git = "https://github.com/KeystoneHQ/rust-bitcoin.git"
-tag = "v0.31.2"
-default-features = false
-features = ["no-std", "secp-recovery"]
+[patch.crates-io]
+bip32 = { git = "https://github.com/KeystoneHQ/crates.git", rev = "9873e8fd56007d792fa60d6e844fdb75d527c858" }
+f4jumble = { git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
+nonempty = { git = "https://github.com/nuttycom/nonempty.git", rev = "38d37189faecb2a0e3d6adc05aa24e1b93c2483b" }
+orchard = { git = "https://github.com/zcash/orchard.git", rev = "e0cc7ac53ad8c97661b312a8b1c064f4cd3c6629" }
+pczt = { git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
+transparent = { package = "zcash_transparent", git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
+zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
+zcash_encoding = { git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
+zcash_keys = { git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
+zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "9407f09208d5574a3ba7bf3e6963741114ba77c2" }
```

Comment:
https://github.com/zcash/librustzcash/pull/1666 has been updated to no longer require a `nonempty` patch (now that it has released a no-std-compatible version):
```suggestion
[patch.crates-io]
bip32 = { git = "https://github.com/KeystoneHQ/crates.git", rev = "9873e8fd56007d792fa60d6e844fdb75d527c858" }
f4jumble = { git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
orchard = { git = "https://github.com/zcash/orchard.git", rev = "cd3e0901ccac2c630dd7fd03eb496d5030c1bbfe" }
pczt = { git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
transparent = { package = "zcash_transparent", git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
zcash_encoding = { git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
zcash_keys = { git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "ab80376bda49e0af524aea23c1d64b3fe1bbe887" }
```